src/cppsim/state.cpp

```diff
@@ -19,6 +19,11 @@ CPPCTYPE inner_product(
 }
```

Comment:
I think `state::inner_product()` should call `state_inner_product_mpi()` when `outer_ac > 0`

Reply:
Right! 
I provided partial support to the inner product in fe2b582.
(Calculating the inner product of the variance and non-dispersion vectors is a TODO.)

Reply:
inner product with multi-cpu is now supported.